src/eval/operation.rs

```diff
@@ -1069,15 +1070,18 @@ fn process_unary_operation(
                         ("index", Term::Num(first_match.start() as f64)),
                         (
                             "groups",
-                            Term::Array(groups, ArrayAttrs { closurized: true })
+                            Term::Array(groups, ArrayAttrs::new().closurized())
                         )
                     )
                 } else {
                     //FIXME: what should we return when there's no match?
                     mk_record!(
                         ("match", Term::Str(String::new())),
                         ("index", Term::Num(-1.)),
-                        ("groups", Term::Array(Vec::new(), Default::default()))
+                        (
+                            "groups",
+                            Term::Array(std::iter::empty().collect(), Default::default())
```

Comment:
Maybe we should have an `Array::new()` constructor, as well as a `Default` impl for arrays?